src/base-command.ts

```diff
@@ -18,6 +28,24 @@ export default abstract class extends Command {
   async init() {
     if (!this.app) {
       this.app = await AppService.create(this.config.configDir);
+
+      if (this.auth_required() && !process.env.ARC_DISABLE_AUTH_CHECK) {
+        const config_dir = this.config.configDir;
```

Comment:
I think this can be simplified to:

```ts
if (this.auth_required && !await this.app.auth.getToken()) {
  this.error(chalk.red(`Please log in using 'architect login'`));
}
```

You'll need to add a `getToken` method to the auth.ts
```ts
async getToken() {
  return await this.credentials.get(`${CREDENTIAL_PREFIX}/token`);
}
```